### PR TITLE
Role show: Added `Admin::RolesController#show` API.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Description

Enable users with the right set of privileges to get details about a role. 

## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3575

**Comment by KH-Amir-TN** [comment]
# REBASED ON 347c40a7263c33cdfd5619c022f88ddae0e29460

File: app/controllers/api/v1/admin/roles_controller.rb [hunk 1]
```diff
@@ -18,6 +18,15 @@ def index
           render_json data: roles
         end
 
+        # GET /api/v1/admin/roles/:role_id.json
+        # Expects: {}
+        # Returns: { data: Array[serializable objects] , errors: Array[String] }
+        # Does: Fetches and returns a role data.
+
+        def show
+          render_json data: @role
```
Comment (KH-Amir-TN) [inline app/controllers/api/v1/admin/roles_controller.rb:27]: @farhatahmad do I create a `RoleSerializer` and use `render_data` in this PR?
Or you're planning it along with the rest of the actions in a seperate PR?

Comment (farhatahmad) [reply app/controllers/api/v1/admin/roles_controller.rb:27]: You can create in another PR if you like

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**